### PR TITLE
Logrotate

### PR DESCRIPTION
**Pull request by clai-ps**
Add logrotate. Default is 5 copies of 20Mb each for each log files.

### CODE REVIEW

File: pureelk.sh [hunk 1]
```diff
@@ -175,9 +179,31 @@ start_containers() {
...
       print_warn "$PUREELK is already running."
   fi
+  
+  print_info "Start Logrotate container..."
+  RUNNING="$(docker inspect -f '{{.State.Running}}' $LOGROTATE)"
+  if [ $? -eq 1 ];
+  then
+      print_warn "$LOGROTATE doesn't exist, starting..."
+      docker run -d -v /var/lib/docker/containers:/var/lib/docker/containers -v /var/log/pureelk:/var/log/pureelk --name $LOGROTATE \
+        -e "LOGS_DIRECTORIES=/var/lib/docker/containers /var/log/pureelk" \
```
Comment (gary-yang) [inline pureelk.sh:189]: Should you also set LOGROTATE_COPIES to a fix number? Otherwise wouldn't we still keep the log files forever?

**Review by gary-yang — CHANGES_REQUESTED** [review]
Looks like we should add LOGROTATE_COPIES setting as well.

**Comment by clai-ps** [comment]
I believe no. Default is 5 copies.

Eric

**Comment by clai-ps** [comment]
This is what I got from my testing:

clai@vm-clai2:/var/log/pureelk$ ls -lh

total 281M

-rw-r--r-- 1 root root 1.9M Jan 26 19:45 array-tasks.log

-rw-r--r-- 1 root root  21M Jan 26 19:43 array-tasks.log.1

-rw-r--r-- 1 root root  21M Jan 26 19:21 array-tasks.log.2

-rw-r--r-- 1 root root  21M Jan 26 19:00 array-tasks.log.3

-rw-r--r-- 1 root root  21M Jan 26 18:39 array-tasks.log.4

-rw-r--r-- 1 root root  21M Jan 26 18:18 array-tasks.log.5

-rw-r--r-- 1 root root  21M Jan 26 17:55 array-tasks.log.6

-rw-r--r-- 1 root root 5.0M Jan 12 11:58 rest.log

-rw-r--r-- 1 root root  20M Jan 11 13:39 rest.log.1

-rw-r--r-- 1 root root 5.8K Jan 10 09:53 rest.log.2

-rw-r--r-- 1 root root 5.8K Jan 10 09:53 rest.log.3

-rw-r--r-- 1 root root 5.8K Jan 10 09:52 rest.log.4

-rw-r--r-- 1 root root 5.8K Jan 10 09:52 rest.log.5

-rw-r--r-- 1 root root 5.8K Jan 10 09:52 rest.log.6

-rw-r--r-- 1 root root  14M Jan 26 19:45 scheduler.log

-rw-r--r-- 1 root root  21M Jan 26 15:32 scheduler.log.1

-rw-r--r-- 1 root root  21M Jan 26 09:16 scheduler.log.2

-rw-r--r-- 1 root root  21M Jan 26 02:59 scheduler.log.3

-rw-r--r-- 1 root root  21M Jan 25 20:43 scheduler.log.4

-rw-r--r-- 1 root root  21M Jan 25 14:26 scheduler.log.5

-rw-r--r-- 1 root root  21M Jan 25 08:10 scheduler.log.6

clai@vm-clai2:/var/log/pureelk$

File: pureelk.sh (hunk 1 above)
Comment (clai-ps) [reply pureelk.sh:189]: We don't need to set it. Default is 5. This is the documentation:
https://github.com/blacklabelops/logrotate/blob/master/README.md

I have tested in my local with the output in the other thread.

**Comment by gary-yang** [comment]
Got it. I ran the container over the weekend. It worked. I'll suggest we specify 10 copies. It stores  about 12 hours of array-tasks.log.

**Comment by clai-ps** [comment]
Hum. But it depends on the number of array that is registered. If you have
a lot of array, it will be less than 12 hours. I can change to 10 if you
think that is the best.

Eric

**Comment by gary-yang** [comment]
Yes, it will depend on the # of arrays. 10 copies is for 12 hours for 1 array in my experiment.

**Comment by clai-ps** [comment]
Or we can set it to 40Mb each file instead of 20Mb. That would have the
same effect.

Eric

**Comment by clai-ps** [comment]
Updated.

**Pull request merged**